### PR TITLE
Add support for additional msvc arguments

### PR DESCRIPTION
**Pull request by majaeger**
Added support for additional parameters. See https://docs.microsoft.com/en-us/cpp/build/reference/compiler-options-listed-alphabetically?view=msvc-170 for more info

ZH
analyze
experimental:deterministic

### CODE REVIEW

File: src/compiler/msvc.rs [hunk 1]
```diff
@@ -516,7 +519,7 @@ pub fn parse_arguments(
                     Argument::Raw(ref val) => {
                         if input_arg.is_some() {
                             // Can't cache compilations with multiple inputs.
-                            cannot_cache!("multiple input files");
+                            cannot_cache!("multiple input files or unknown argument: ", String::from(input_arg.unwrap().to_string_lossy()));
```
Comment (drahnr) [inline src/compiler/msvc.rs:522]: Please use an `if let Some(input_arg) = input_arg { .. }` to avoid the `unwrap`

File: src/compiler/msvc.rs [hunk 2]
```diff
@@ -378,6 +379,7 @@ msvc_args!(static ARGS: [ArgInfo<ArgData>; _] = [
     msvc_flag!("Zp4", PassThrough),
     msvc_flag!("Zp8", PassThrough),
     msvc_flag!("Zs", SuppressCompilation),
+    msvc_flag!("analyze", PassThrough),
```
Comment (drahnr) [inline src/compiler/msvc.rs:382]: Cross checked, https://docs.microsoft.com/en-us/cpp/build/reference/compiler-options-listed-alphabetically?view=msvc-170

**Review by drahnr — CHANGES_REQUESTED** [review]
Small nit, otherwise LGTM!

File: src/compiler/msvc.rs (hunk 2 above)
Comment (drahnr) [reply src/compiler/msvc.rs:382]: Jesus, pre coffee I should not be allow on the internet.

File: src/compiler/msvc.rs (hunk 1 above)
Comment (majaeger) [reply src/compiler/msvc.rs:522]: Nice. I'm new to rust and didn't know about this!

File: src/compiler/msvc.rs [hunk 3]
```diff
@@ -516,7 +519,11 @@ pub fn parse_arguments(
                     Argument::Raw(ref val) => {
                         if input_arg.is_some() {
```
Comment (drahnr) [inline src/compiler/msvc.rs:520]:
```suggestion
                        if let Some(ref input_arg) = input_arg {
```

File: src/compiler/msvc.rs [hunk 4]
```diff
@@ -516,7 +519,11 @@ pub fn parse_arguments(
...
                             // Can't cache compilations with multiple inputs.
-                            cannot_cache!("multiple input files");
+                            if let Some(input_arg) = input_arg {
+                                cannot_cache!("multiple input files or unknown argument: ", String::from(input_arg.to_string_lossy()));
+                            } else {
+                                cannot_cache!("multiple input files");
+                            }
```
Comment (drahnr) [inline src/compiler/msvc.rs:526]:
```suggestion
                            cannot_cache!("multiple input files or unknown argument: ", input_arg.to_string_lossy());
```

**Review by drahnr — APPROVED** [review]
Thanks!

**Comment by drahnr** [comment]
> Sorry for the false start. Looks like I still needed the String::from() call

My bad, remembered the return value wrongly and should have double checked.

**Review by drahnr — APPROVED** [review]

**Pull request merged**